engine/src/main/java/org/terasology/rendering/dag/nodes/VignetteNode.java

```diff
@@ -0,0 +1,157 @@
+import org.terasology.rendering.dag.AbstractNode;
+import org.terasology.rendering.dag.StateChange;
+import org.terasology.rendering.dag.stateChanges.BindFbo;
+import org.terasology.rendering.dag.stateChanges.SetViewportToSizeOf;
+import org.terasology.rendering.dag.stateChanges.SetInputTextureFromFbo;
+import org.terasology.rendering.dag.stateChanges.EnableMaterial;
+import org.terasology.rendering.dag.stateChanges.SetInputTexture2D;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.ScreenGrabber;
+import org.terasology.rendering.opengl.fbms.DisplayResolutionDependentFBOs;
+import org.terasology.rendering.world.WorldRenderer;
+import org.terasology.world.WorldProvider;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+import static org.terasology.rendering.opengl.fbms.DisplayResolutionDependentFBOs.FINAL_BUFFER;
+import static org.terasology.rendering.dag.nodes.FinalPostProcessingNode.POST_FBO_URI;
+
+import static org.terasology.rendering.dag.stateChanges.SetInputTextureFromFbo.FboTexturesTypes.ColorTexture;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+
+/**
+ * An instance of this node adds vignette onto the rendering achieved so far, stored in the gbuffer.
+ * It should provide ability to use various vignette textures and tinting.
+ * 1 Channeled transparency texture is used atm. Furthermore, depending if a screenshot has been requested,
+ * it instructs the ScreenGrabber to save it to a file.
+ * Stores the result into the displayResolutionDependentFBOs.FINAL_BUFFER, to be displayed on the screen.
+ * Requirements: https://github.com/MovingBlocks/Terasology/issues/3040
+ */
+public class VignetteNode extends AbstractNode implements PropertyChangeListener {
+    private static final ResourceUrn VIGNETTE_MATERIAL_URN = new ResourceUrn("engine:prog.vignette");
+
+    private RenderingConfig renderingConfig;
+    private WorldProvider worldProvider;
+    private WorldRenderer worldRenderer;
+    private SubmersibleCamera activeCamera;
+
+    private ScreenGrabber screenGrabber;
+
+    private Material vignetteMaterial;
+
+    private boolean vignetteIsEnabled;
+    // TODO: figure where from to set this variable
+    private Vector3f tint = new Vector3f(.0f, .0f, .0f);
+//    private Vector3f tint = new Vector3f(0.937f,0.126f,0.016f);
+
+    private StateChange setVignetteInputTexture;
+
+    public VignetteNode(String nodeUri, Context context) {
+        super(nodeUri, context);
+
+        worldProvider = context.get(WorldProvider.class);
+
+        worldRenderer = context.get(WorldRenderer.class);
+        activeCamera = worldRenderer.getActiveCamera();
+        screenGrabber = context.get(ScreenGrabber.class);
+
+        DisplayResolutionDependentFBOs displayResolutionDependentFBOs = context.get(DisplayResolutionDependentFBOs.class);
+        // TODO: see if we could write this straight into a GBUFFER
```

Comment:
Why do you want to write to GBUFFER, instead of FINAL_BUFFER? Regardless, it then has to be copied to the screen.

Reply:
This comment is actually just a copy paste from another node. I haven't gone trough every line yet because there are some things still open